pyimzml/ImzMLWriter.py

```diff
@@ -123,43 +133,69 @@ def __init__(self, output_filename, mz_dtype=np.float64, intensity_dtype=np.floa
         self.imzml_template = self.wheezy_engine.get_template('imzml')
 
         self.spectra = []
+        self.hashes = {} #set this to None to force processed mode.
```

Comment:
I would choose the last option and I don't see it as very complicated, feel free to  use something like LRU cache decorators from http://pythonhosted.org/cachetools/